tox/config.py

```diff
@@ -913,8 +913,20 @@ def _split_env(env):
 
 
 def _split_factor_expr(expr):
+    def split_single(e):
+        raw = e.split('-')
+        included = {x for x in raw if x[:1] != '!'}
```

Comment:
this is a bit cryptic, consider instead using ``not startswith`` to make it more explicit